flink-python/pyflink/common/typeinfo.py

```diff
@@ -407,12 +406,12 @@ def to_internal_type(self, obj):
                 raise ValueError("Unexpected tuple %r with RowTypeInfo" % obj)
         else:
             if isinstance(obj, dict):
-                return tuple(obj.get(n) for n in self._j_typeinfo.getFieldNames())
+                return tuple(obj.get(n) for n in self.field_names)
```

Comment:
```suggestion
                return tuple(obj[n] for n in self.field_names)
```

Reply:
Here we should appy obj.get(field_name), it would return None if the field_name does not present in the obj while applying obj[field_name] will get a KeyError.